docs/azure-functions.asciidoc

```diff
@@ -0,0 +1,155 @@
+:framework: Azure Functions
+
+[[azure-functions]]
+
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/nodejs/current/azure-functions.html[elastic.co]
+endif::[]
+
+=== Monitoring Node.js Azure Functions
+
+The Node.js APM Agent can be used trace function invocations in an https://learn.microsoft.com/en-us/azure/azure-functions/[Azure Functions] app.
+
+
+[float]
+[[azure-functions-prerequisites]]
+==== Prerequisites
+
+You need an APM Server to send APM data to. Follow the
+{apm-guide-ref}/apm-quick-start.html[APM Quick start] if you have not set one up
+yet. You will need your *APM server URL* and an APM server *secret token* (or
+*API key*) for configuring the APM agent below.
+
+You will also need an Azure Function app to monitor. If you do not have an
+existing one, you can follow https://learn.microsoft.com/en-us/azure/azure-functions/create-first-function-cli-node#create-supporting-azure-resources-for-your-function[this Azure guide]
+to create one.
+
+[IMPORTANT]
+====
+If you use `func init --javascript ...` as suggested in this Azure guide,
+then it is recommended that you *uninstall* the `azure-functions-core-tools`
+dependency by running `npm uninstall azure-functions-core-tools` and
+https://github.com/Azure/azure-functions-core-tools#installing[install it separately].
+Having `azure-functions-core-tools` as a "devDependency" in your package.json
+will result in unreasonably large deployments that will be very slow to publish
+and will run your Azure Function app VM out of disk space.)
+====
+
+You can also take a look at and use this https://github.com/elastic/apm-agent-nodejs/tree/main/examples/an-azure-function-app/[Azure Functions example app with Elastic APM already integrated].
+
+[float]
+[[azure-functions-setup]]
+==== Step 1: Add the APM agent dependency
+
+Add the `elastic-apm-node` module as a dependency of your application:
+
+[source,bash]
+----
+npm install elastic-apm-node --save  # or 'yarn add elastic-apm-node'
+----
+
+
+[float]
+==== Step 2: Start the APM agent
+
+For the APM agent to instrument Azure Functions, it needs to be started when the
+Azure host starts its Node.js worker processes. The best way to do so is by
+using an app-level entry point (support for this was added for Node.js Azure
+Functions https://github.com/Azure/azure-functions-nodejs-worker/issues/537[here]).
+
+1. Create a module to start the APM agent. For example, a file at the root of your repository named "initapm.js":
++
+[source,javascript]
+----
+// initapm.js
+require('elastic-apm-node').start({
+  <1>
+})
+----
+<1> Optional <<configuration,configuration options>> can be added here.
+
+2. Add a "main" entry to your package.json pointing to the app init file.
++
+[source,json]
+----
+...
+  "main": "initapm.js",
+...
+----
++
```

Comment:
Honestly, I'm impressed by your `+` usage to nest the next sentence into step two.

Reply:
https://docs.asciidoctor.org/asciidoc/latest/lists/continuation/#list-continuation  :flex: :)